sda-commons-server-auth/src/main/java/org/sdase/commons/server/opa/filter/OpaAuthFilter.java

```diff
@@ -131,7 +130,14 @@ public void filter(ContainerRequestContext requestContext) {
         // process the actual request to the open policy agent server
         String[] path =
             uriInfo.getPathSegments().stream().map(PathSegment::getPath).toArray(String[]::new);
-        OpaRequest request = OpaRequest.request(jwt, path, method, trace, headers);
+        OpaInput opaInput = new OpaInput(jwt, path, method, trace);
+        ObjectNode objectNode = om.convertValue(opaInput, ObjectNode.class);
+
+        // append the input extensions to the input object
+        inputExtensions.forEach(
+            ex -> objectNode.set(ex.getNamespace(), ex.filter(requestContext, om)));
```

Comment:
Maybe we need a test and document how implementations can deal with `RequestContext#getEntityStream()`. It may be the case that it is broken after once read. This happens regular when one is trying to implement a debug request logger that logs the request or response body.

We may have to wrap the `RequestContext` and throw an Exception or copy and reset the `InputStream` when `getEntityStream()` is called. maybe we can do the latter in a second PR. In that case we have to watch effects on memory consumption.

Reply:
As discussed outside of GitHub we should also document drawbacks when query params are used, e.g. to map all occurences of a query param as List in the endpoints.

Also the header authorization has to be used carefully because services often only use the first occurrence of a header.

Reply:
I'll add a test that shows this behavior but don't fix the issue in this PR.